src/katgpucbf/fgpu/engine.py

```diff
@@ -1405,20 +1405,37 @@ async def _run_receive(self, streams: list[spead2.recv.ChunkRingStream], layout:
         for pipeline in self._pipelines:
             pipeline.shutdown()
 
-    async def request_gain(self, ctx, input: int, *values: str) -> tuple[str, ...]:
+    def _request_pipeline(self, stream: str) -> Pipeline:
+        """Find the pipeline related to a katcp request.
+
+        Raises
+        ------
+        FailReply
+            If the `stream` is not a known output.
+        """
+        # Note: this takes O(n) time, but as there are expected to be less than 10
+        # pipelines it is probably not worth keeping a dictionary.
```

Comment:
Maybe it's my lack of fresh memory on requirements (or some other missed discussion), but how are you sure that
> [...] there are expected to be less than 10 pipelines [...] (?)